Eastern-Research-Group/mywaterway
# Feature/426 integrate surrounding waterbodies monitoring

## Related Issues:
* [HMW-426](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=864&view=detail&selectedIssue=HMW-426)

## Main Changes:
* Updated the `useAllFeaturesLayer` hook to employ separate layers for "local" and surrounding data, rather than selectively masking features.
* Refactored the `monitoringLocationsLayer` to use the `useAllFeaturesLayer` hook, combining it with the `surroundingMonitoringLocationsLayer`.
* Integrated the `monitoringLocationsLayer` and the `allWaterbodiesLayer` into the aggregate "Surrounding Features" widget.
    * I held off on refactoring/combining the waterbodies layers for now, and instead integrated the `allWaterbodiesLayer` as is. This is so the EPA team could start testing the functionality already in place, but it results in some less-than-ideal workarounds: the `waterbodyLayer` key is duplicated in the _Layers_ context, and renamed as needed, since the `SurroundingsWidget` compares the keys to those existing in `visibleLayers` to determine whether to show the layer in the widget. This also means that it was necessary to add `waterbodyLayer` to the `visibleLayers` object on the _Identified Issues_ tab (hidden from the list).
* Fixed an issue of dischargers not filtering correctly on the _Identified Issues_ tab.
* Removed the `BoundariesToggleLayer` class, it's sub-classes, and the `experimentalDecorators` allowance in `tsconfig.json`. 

## Steps To Test:
1. Compare the default visibility and functionality of the surrounding monitoring locations and waterbodies to the separate widgets in the production app, and verify they match.
2. Since the `monitoringLocationsLayer` was refactored pretty heavily, check that both its interior and surrounding features update as expected, especially on the _Monitoring_ tab, on the _Tribal_ pages, and on the _Monitoring Report_ pages.
3. Check that the "Surrounding Features" widget is not visible on the State _Advanced Search_ page map.
4. Navigate to http://localhost:3000/community/dc/identified-issues and check that dischargers are filtered correctly on the map and in the "Dischargers with Significant Violations" list. A single facility should be visible on the base layer, but the other HUC facilities and surrounding facilities should be visible when "Dischargers" is toggled on in the "Surrounding Features" widget.
